packages/big-design/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,104 @@
+import { CheckIcon } from '@bigcommerce/big-design-icons';
+import React, { HTMLAttributes, memo, useEffect, useState } from 'react';
+
+import {
+  StyledDash,
+  StyledLight,
+  StyledLightText,
+  StyledMobileLight,
+  StyledMobileLightText,
+  StyledStep,
+  StyledStepper,
+  StyledText,
+} from './styled';
+
+export interface StepProps {
+  number: number;
+  text?: string;
+  state?: 'pending' | 'current' | 'done';
+  isLast?: boolean;
+}
+
+export interface StepperProps extends HTMLAttributes<HTMLDivElement> {
+  steps: Array<string>;
+  currentStep: number;
+}
+
+export interface LightProps {
+  active?: boolean;
+}
+export interface DashProps {
+  active?: boolean;
+}
+export interface TextProps {
+  active?: boolean;
+}
+
+export const Step: React.FC<StepProps> = ({ number, text, state, isLast }) => {
+  const [isCurrent, setIsCurrent] = useState(state === 'current');
+  const [isDone, setIsDone] = useState(state === 'done');
+  useEffect(() => {
+    setIsCurrent(state === 'current');
+    setIsDone(state === 'done');
+  }, [state]);
+
+  return (
+    <StyledStep>
```

Comment:
Couple of things with this component:
1. I think using a `Flex` is better that using a grid. According to the designs, each step should be the same length as the others and using that grid `minmax(6rem, 14.25rem)` logic will cause them to collapse non-uniformly on smaller screen sizes.
2. This is a big miss – we need to add in a11y props. I added them [here](https://github.com/Spawbreaker/big-design/compare/stepper...chanceaclark:stepper-updates#diff-416e0389ee27a965342469d03210e69c56a892b1a4af9141a6b06f31272eff81R16-R23). Here is also the https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role